webpack/assets/javascripts/react_app/components/Editor/EditorActions.js

```diff
@@ -0,0 +1,127 @@
+import API from '../../API';
+import { translate } from '../../common/I18n';
+
+import {
+  EDITOR_INITIALIZE,
+  EDITOR_CHANGE_STATE,
+  EDITOR_IMPORT_FILE,
+  EDITOR_REVERT_CHANGES,
+  EDITOR_EXEC_PREVIEW,
+  EDITOR_MODAL_TOGGLE,
+} from './EditorConstants';
+
+export const initializeEditor = props => dispatch => {
+  const {
+    data: { template, locked, type, hosts },
+    readOnly,
+    isMasked,
+    selectedView,
+    isRendering,
+    renderedValue,
+    showError,
+  } = props;
+
+  const initialState = {};
+
+  // initialize after changing editors
+  initialState.value = template || '';
+  if (readOnly !== locked) {
+    if (locked === true) initialState.readOnly = true;
+    else initialState.readOnly = false;
+  }
+  if (isMasked && type === 'templates') initialState.isMasked = false;
+  if (selectedView !== 'input') initialState.selectedView = 'input';
+  if (isRendering) initialState.isRendering = false;
+  if (renderedValue !== '') initialState.renderedValue = '';
+  if (showError) initialState.showError = false;
+
+  if (hosts && hosts.length !== 0) {
+    initialState.hosts = Object.assign(
+      {},
+      ...hosts.map(element => ({ [element.host.name]: element.host.id }))
+    );
```

Comment:
What's the difference between:

```js
initialState.hosts = {
  ...hosts.map(element => ({ [element.host.name]: element.host.id }))
};
```

Reply:
This won't work, as it returns an object of objects e.g.
```
 {
   0: {aaron-ellert.ramon-eatherly.gail-walema.example.org: 22}
   1: {abel-bearman.ruben-montieth.example.net: 66}
   2: {alvin-bocchini.erika-kirton.elton-lucore.example.org: 44}
 }
```

my code returns:

```
 {
   aaron-ellert.ramon-eatherly.gail-walema.example.org: 22
   abel-bearman.ruben-montieth.example.net: 66
   alvin-bocchini.erika-kirton.elton-lucore.example.org: 44
 }
```